website/client/js/controllers/filtersCtrl.js

```diff
@@ -25,6 +25,9 @@ habitrpg.controller("FiltersCtrl", ['$scope', '$rootScope', 'User', 'Shared',
     };
 
     $scope.toggleFilter = function(tag) {
+      if (user.filters === undefined) {
```

Comment:
the PR looks good! Maybe `if (!user.filter) user.filters = {}` should be added here https://github.com/HabitRPG/habitrpg/blob/develop/website/client/js/services/userServices.js#L82 right after `_extend` instead than in `toggleFilter`. So if something else relies on `user.filters` it will work even if `toggleFilter` is not called before